### DISCUSSION TITLE
Obsolete and conflicts same package

### DISCUSSION THREAD

**Discussion by AliyevH** [rpm-software-management/rpm#2225] [Q&A]
How to correctly use obsolete and conflict interchangeably.

For example

```
%package libs

Summary: The shared library
Conflicts: mariadb-connector-c-config
Conflicts: mariadb-connector-c


%package -n meta
Summary:  meta package
Requires: %{name}-libs
Provides:  mariadb-connector-c = 3.2.8-1 mariadb-connector-c-config = 3.2.8-1
Obsoletes:  mariadb-connector-c mariadb-connector-c-config
```
In such situation which way is more accurate?

**Comment by pmatilai** [comment]
Obsoletes and conflicts are not interchangeable, they serve very different purposes.
Obsoletes X means this package replaces X from here to eternity and generally must not be used if X continues to exist. Conflicts merely prevent parallel installation.

**Comment by AliyevH** [comment]
@pmatilai Thanks for response.
in this situation i don't even need conflicts directive for package if i am going to obsolete and provide mariadb-connector-c. correct?
